pkgs/applications/misc/pdfstudio/default.nix

```diff
@@ -1,87 +1,42 @@
-{ stdenv
-, lib
+{ program ? "pdfstudioviewer"
 , fetchurl
 , libgccjit
-, dpkg
-, makeDesktopItem
-, copyDesktopItems
-, autoPatchelfHook
-, sane-backends
-, jdk11
+, callPackage
 }:
 
-# See also package 'pdfstudioviewer'
-# Differences are ${pname}, Download directory name (PDFStudio / PDFStudioViewer),
-# sha256, and libgccjit (not needed for PDFStudioViewer)
-let year = "2021";
-in
-stdenv.mkDerivation rec {
-  pname = "pdfstudio";
-  version = "${year}.1.2";
-  strictDeps = true;
-
-  src = fetchurl {
-    url = "https://download.qoppa.com/${pname}/v${year}/PDFStudio_v${
+{
+  pdfstudio = callPackage ./common.nix rec {
+    pname = program;
+    year = "2021";
+    version = "${year}.1.2";
+    desktopName = "PDF Studio";
+    longDescription = ''
+    PDF Studio is an easy to use, full-featured PDF editing software. This is the standard/pro edition, which requires a license. For the free PDF Studio Viewer see the package pdfstudioviewer.
+    '';
+    extraBuildInputs = [
+      libgccjit #for libstdc++.so.6 and libgomp.so.1
+    ];
+    src = fetchurl {
+      url = "https://download.qoppa.com/${pname}/v${year}/PDFStudio_v${
```

Comment:
Maybe a small function that returns the url for both programs might be useful, since they pretty much share the same url. It looks a bit repetitive to me now, with how many lines it takes

That's in no way a blocker for this to be merged though, i think it would just be nice.